src/ApiCore/AuthWrapper.php

```diff
@@ -55,6 +61,54 @@ public function __construct(FetchAuthTokenInterface $fetchAuthTokenInterface, ca
         $this->authHttpHandler = $authHttpHandler;
     }
 
+    /**
+     * Factory method to create an AuthWrapper for a given set of scopes. If no credentials are
+     * provided in $keyFile or $keyFilePath, then ApplicationDefaultCredentials will be used.
+     *
+     * @param string[] $scopes The scopes required by this AuthWrapper.
+     * @param array $args {
+     *     @type string $keyFile
+     *           Optional. JSON credentials as an associative array.
+     *     @type string $keyFilePath
+     *           Optional. A JSON credential file path. If $keyFile is specified, $keyFilePath is ignored.
+     *     @type CacheItemPoolInterface $authCache
+     *           Optional. A cache for storing access tokens. Defaults to a simple in memory implementation.
+     * }
+     * @return AuthWrapper
+     * @throws ValidationException
+     */
+    public static function createWithScopes(array $scopes, array $args = [])
+    {
+        $args += [
+            'keyFile' => null,
+            'keyFilePath' => null,
+            'authCache' => null,
+        ];
+
+        $keyFile = $args['keyFile'] ?: $args['keyFilePath'];
+        $authCache = $args['authCache'] ?: new MemoryCacheItemPool();
+
+        try {
+            $authHttpHandler = HttpHandlerFactory::build();
+        } catch (\Exception $ex) {
+            throw new ValidationException("Failed to create authHttpHandler", $ex->getCode(), $ex);
+        }
+
+        if (is_null($keyFile)) {
+            $loader = ApplicationDefaultCredentials::getCredentials($scopes, $authHttpHandler);
+        } else {
+            $loader = new ServiceAccountCredentials($scopes, $keyFile);
```

Comment:
This looks like we should be using [`CredentialsLoader::makeCredentials()`](https://github.com/google/google-auth-library-php/blob/master/src/CredentialsLoader.php#L115)

Reply:
Done